lib/spack/spack/spec.py

```diff
@@ -753,35 +754,83 @@ def __str__(self):
         return "{deps: %s}" % ', '.join(str(d) for d in sorted(self.values()))
 
 
-def _libs_default_handler(descriptor, spec, cls):
-    """Default handler when looking for 'libs' attribute. The default
-    tries to search for 'lib{spec.name}' recursively starting from
-    `spec.prefix`.
+def _executable_default_handler(descriptor, spec, cls):
+    """Default handler when looks for 'executable' attribute. The default
+    tries to search for {spec.name} in the ``spec.prefix.bin`` directory.
 
     :param ForwardQueryToPackage descriptor: descriptor that triggered
         the call
     :param Spec spec: spec that is being queried
     :param type(spec) cls: type of spec, to match the signature of the
         descriptor `__get__` method
     """
-    name = 'lib' + spec.name
-    shared = '+shared' in spec
-    return find_libraries(
-        name, root=spec.prefix, shared=shared, recurse=True
-    )
+    exe = os.path.join(spec.prefix.bin, spec.name)
+
+    if os.path.isfile(exe) and os.access(exe, os.X_OK):
+        return exe
+    else:
+        msg = 'Unable to locate {0} executable in {1}'
+        raise RuntimeError(msg.format(spec.name, spec.prefix.bin))
+
+
+def _headers_default_handler(descriptor, spec, cls):
+    """Default handler when looking for 'headers' attribute. The default
+    tries to search for ``*.h`` files recursively starting from
+    ``spec.prefix.include``.
+
+    :param ForwardQueryToPackage descriptor: descriptor that triggered
+        the call
+    :param Spec spec: spec that is being queried
+    :param type(spec) cls: type of spec, to match the signature of the
+        descriptor `__get__` method
+    """
+    headers = find_headers('*', root=spec.prefix.include, recurse=True)
+
+    if headers:
+        return headers
+    else:
+        msg = 'Unable to locate {0} headers in {1}'
+        raise RuntimeError(msg.format(spec.name, spec.prefix.include))
 
 
-def _cppflags_default_handler(descriptor, spec, cls):
-    """Default handler when looking for cppflags attribute. The default
-    just returns '-I{spec.prefix.include}'.
+def _libs_default_handler(descriptor, spec, cls):
+    """Default handler when looking for 'libs' attribute. The default
+    tries to search for 'lib{spec.name}' recursively starting from
+    ``spec.prefix``.
 
     :param ForwardQueryToPackage descriptor: descriptor that triggered
         the call
     :param Spec spec: spec that is being queried
     :param type(spec) cls: type of spec, to match the signature of the
         descriptor `__get__` method
     """
-    return '-I' + spec.prefix.include
+    name = 'lib' + spec.name
```

Comment:
I wonder if we should just search for all libraries instead of just `lib{name}`. My default header handler searches for all `*.h` files.